drivers/sensor/max30208/max30208.c

```diff
@@ -0,0 +1,346 @@
+ */
+
+#define DT_DRV_COMPAT maxim_max30208
+
+#include <sys/byteorder.h>
+#include <logging/log.h>
+
+#include "max30208.h"
+
+LOG_MODULE_REGISTER(MAX30208, CONFIG_SENSOR_LOG_LEVEL);
+
+static int max30208_start_measurement(const struct device *dev)
+{
+	const struct max30208_config *config = dev->config;
+
+	/* Trigger one measurement */
+	int ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_TEMP_SETUP,
+					MAX30208_CONVERT_T_MASK);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int max30208_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+	struct max30208_data *data = dev->data;
+	uint8_t local_buffer[MAX30208_BYTES_PER_VAL] = { 0 };
+
+	/* If new value in ring buffer available, get it */
+	if (ring_buf_size_get(&data->raw_buffer) < MAX30208_BYTES_PER_VAL) {
+		return -ENODATA;
+	}
+
+	int ret = ring_buf_get(&data->raw_buffer, local_buffer, MAX30208_BYTES_PER_VAL);
+
+	if (ret < 0) {
+		LOG_ERR("Couldn't get data from ringbuffer");
+	}
+
+	/* Concat two bytes to a int16 value */
+	int16_t temp_val = (int16_t)sys_get_be16(local_buffer);
+
+	/* Convert raw value into temperature */
+	val->val1 = temp_val / MAX30208_ONE_DEGREE;
+	val->val2 = (temp_val % MAX30208_ONE_DEGREE) * MAX30208_LSB_e6;
+
+	return 0;
+}
+
+#ifndef CONFIG_MAX30208_TRIGGER
+static int max30208_data_ready(const struct device *dev, uint8_t *ready)
+{
+	const struct max30208_config *config = dev->config;
+	struct max30208_status status;
+
+	/* Read data ready */
+	int ret =
+		i2c_reg_read_byte_dt(&config->bus, MAX30208_REG_INT_STS, MAX30208_STC_TO_P(status));
+
+	if (ret < 0) {
+		LOG_ERR("Timeout while waiting on MAX30208 data ready\n");
+		return ret;
+	}
+
+	*ready = status.temp_rdy;
+
+	return 0;
+}
+#endif
+
+int max30208_readout_batch(const struct device *dev)
+{
+	const struct max30208_config *config = dev->config;
+	struct max30208_data *data = dev->data;
+
+	uint8_t fifo_data_counter = 0;
+	int ret =
+		i2c_reg_read_byte_dt(&config->bus, MAX30208_REG_FIFO_DATA_CTR, &fifo_data_counter);
+
+	if (ret < 0) {
+		LOG_ERR("Timeout while waiting on FIFO data counter\n");
+		return ret;
+	}
+
+	/* Read data ready */
+	uint8_t local_buffer[MAX30208_BYTES_PER_VAL * MAX30208_FIFO_SIZE];
+
+	ret = i2c_burst_read_dt(&config->bus, MAX30208_REG_FIFO_DATA, local_buffer,
+				fifo_data_counter * MAX30208_BYTES_PER_VAL);
+
+	if (ret < 0) {
+		LOG_ERR("Could not batch read sensor values");
+		return ret;
+	}
+
+	/* Put both bytes into ring buffer */
+	for (uint8_t i = 0; i < fifo_data_counter; i++) {
+		if (ring_buf_space_get(&data->raw_buffer) < MAX30208_BYTES_PER_VAL) {
+			struct sensor_value lost_val;
+
+			ret = max30208_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &lost_val);
+			if (ret < 0) {
+				return ret;
+			}
+			LOG_INF("Buffer size to small. Value %d.%d is lost", lost_val.val1,
+				lost_val.val2);
```

Comment:
is it worth reading just to log? what if logging is disabled? Isn't it sufficient to warn about buffer being too small?

Reply:
since one item must be poped on a full ringbuffer, I could just pop and not store it, but I don't see a lot of advantage here..